perllib/FixMyStreet/App/Controller/Report/New.pm

```diff
@@ -776,6 +776,11 @@ sub setup_categories_and_bodies : Private {
         $bodies_to_list{ $contact->body_id } = $contact->body;
 
         my $metas = $contact->get_metadata_for_input;
+
+        for (@$metas) {
+            $_->{code} =~ s/\s/_/g if $_->{code};
```

Comment:
GitHub lost my comment, sigh.
I'm afraid this won't work - you're changing what is output, but then what is read back in in both `check_for_category` and `set_report_extras` does not know to look for this changed code and so won't be reading in the information.

I don't think this is the correct place to try and fix this issue. The possibilities I can see are:
1. Find and fix the JS that is erroring because it can't handle a code with a space in, in a similar way to ed0378056d1d3445a0d7b321da3652aa17f12292
2. In the admin, when someone submits a code with a space in, change it to underscore at that point - so it's stored in the database like that and is consistent everywhere
3. In the admin, error when someone submits a code with a space in and make them change it (and perhaps use a pattern attribute so they can't type a space in the box).

Not sure which is "best" of those - 1 would mean people could still have codes with spaces in, but do we really want that, doesn't feel like something that's ever needed; 2 would work fine, probably won't confuse anyone if it should happen; 3 might be most clear/straightforward? But happy to hear suggestions if you think another is preferable.

Reply:
I went with a version of 3) in the end. Maybe I should have just gone with an HTML pattern attribute but I felt a deeper check would be 'safer' (e.g. does the pattern attribute work on all browsers? Could it possibly fail/be overridden in some way? Though I may be overthinking it).